corehq/apps/cleanup/management/commands/swap_duplicate_xforms.py

```diff
@@ -0,0 +1,158 @@
+import re
+from collections import defaultdict
+from optparse import make_option
+
+from datetime import datetime
+from uuid import UUID
+
+from django.core.management import BaseCommand
+
+from couchforms.models import XFormInstance, XFormDuplicate
+
+
+class Command(BaseCommand):
+    help = 'Replace xforms missing attachments with xfrom duplicates containing attachments.'
+    args = '<ids_file_path> <log_path>'
+
+    option_list = BaseCommand.option_list + (
+        make_option(
+            '--dry-run',
+            action='store_true',
+            dest='dry_run',
+            default=False,
+            help="Don't do the actual modifications, but still log what would be affected."
+        ),
+        make_option(
+            '--no-input',
+            action='store_true',
+            dest='no_input',
+            default=False,
+            help='Skip important confirmation warnings.'
+        ),
+    )
+
+    def handle(self, *args, **options):
+
+        self.dups_by_domain = {}
+        # e.g.
+        #   {
+        #       "foo-domain": {
+        #           "an-original-id": {"a-duplicate-id", "another-duplicate-id"}
+        #       }
+        #   }
+
+        dry_run = options.get("dry_run", True)
+        if not options.get('no_input', False) and not dry_run:
+            confirm = raw_input(
+                u"""
+                Are you sure you want to swap the given XFormInstances for duplicates?
+                This is NOT a dry run. y/N?
+                """
+            )
+            if confirm != "y":
+                print "\n\t\tSwap duplicates cancelled."
+                return
+
+        ids_file_path = args[0].strip()
+        log_path = args[1].strip()
+
+        num_with_1_dup = 0
+        num_with_multi_dup = 0
+        num_with_no_dup = 0
+
+        with open(log_path, "w") as log_file:
+            with open(ids_file_path, "r") as lines:
+
+                total_lines = 0
+                for _ in lines:
+                    total_lines += 1
+                lines.seek(0)
+
+                for i, (domain, bad_xform_id) in enumerate(l.split() for l in lines):
+                    self._print_progress(i, total_lines)
+                    duplicates = self.get_duplicates(domain, bad_xform_id)
+                    if len(duplicates) == 1:
+                        num_with_1_dup += 1
+                        self.swap_doc_types(log_file, bad_xform_id, list(duplicates)[0], dry_run)
+                    elif len(duplicates) > 1:
+                        num_with_multi_dup += 1
+                        self.log_too_many_dups(log_file, bad_xform_id, duplicates)
+                    else:
+                        num_with_no_dup += 1
+
+        print "Found {} forms with no duplicates".format(num_with_no_dup)
+        print "Found {} forms with one duplicate".format(num_with_1_dup)
+        print "Found {} forms with multiple duplicates".format(num_with_multi_dup)
```

Comment:
Breakdown when running with the id file Cal gave me is:
```
Found 1191 forms with no duplicates
Found 358 forms with one duplicate
Found 6 forms with multiple duplicates
```
Cal mentioned there could be some false positives, but this seems a little on the high side. Just want to confirm that this seem plausible @calellowitz?